### PR TITLE
refactor: dir package [1]

### PR DESCRIPTION
**Pull request by JeyJeyGao**
```go
// The relative path to {NOTATION_CONFIG}
const (
    PathConfigFile  = "config.json"
    PathSigningKeys = "signingkeys.json"
    PathTrustPolicy = "trustpolicy.json"
)

func LocalKeyPath(name string) (keyPath, certPath string)
func X509TrustStoreDir(items... string) string

var (
    UserConfigDir    string // Aboslute path of user level {NOTATION_CONFIG}
    UserLibexecDir   string // Aboslute path of user level {NOTATION_LIBEXEC}
)

// ConfigFS and PluginFS can be an union fs
func ConfigFS() SysFS
func PluginFS() SysFS

type SysFS interface {
    fs.FS

    SysPath(items ...string) (string, error)
}
    func NewSysFS(root string) SysFS
```

Please review this PR before others.
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: internal/mock/mock_fs/fs.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package mock_fs
```
Comment (shizhMSFT) [inline internal/mock/mock_fs/fs.go:1]: Since it is under `mock` folder, `package mockfs` or `package fs` is more appropriate.

File: dir/fs.go [hunk 2]
```diff
@@ -1,156 +1,50 @@
...
-// NewRootedFS create a rootedFS
+// NewSysFS returns the SysFS for the given roots directories.
 //
-// if fsys is nil, uses os.DirFS
-func NewRootedFS(root string, fsys fs.FS) RootedFS {
-	if fsys == nil {
-		return &rootedFS{FS: os.DirFS(root), root: root}
+// Support one root directory for rc.1, and may support union directories FS
+// after rc.1.
+func NewSysFS(roots ...string) SysFS {
```
Comment (shizhMSFT) [inline dir/fs.go:33]: How about 
```go
func NewSysFS(root string) SysFS
```

For "union directories FS after rc.1", maybe we can have something like
```golang
func NewUnionFS(roots ...string) (SysFS, error)
```

File: dir/path.go [hunk 3]
```diff
@@ -1,88 +1,71 @@
...
+	SystemLibexecDir string // Absolute path of system level {NOTATION_LIBEXEC}
+	UserConfigDir    string // Absolute path of user level {NOTATION_CONFIG}
+	UserLibexecDir   string // Absolute path of user level {NOTATION_LIBEXEC}
+)
 
-	// LocalCertificateExtension defines the extension of the certificate files
+// The relative path to {NOTATION_CONFIG}
+const (
+	// notation is the directory name for notation configurations.
+	notation = "notation"
```
Comment (shizhMSFT) [inline dir/path.go:19]: Let's move this unexported const to a separated `const` block.

File: dir/fs.go [hunk 4]
```diff
@@ -1,156 +1,50 @@
...
-		dirs = append(dirs, filepath.Join(userLibexec, "plugins"))
-		dirs = append(dirs, filepath.Join(systemLibexec, "plugins"))
-	}
-	for _, dir := range dirs {
-		rootedFsys = append(rootedFsys, NewRootedFS(dir, nil))
-	}
-	return NewUnionDirFS(rootedFsys...)
+// PluginFS is the plugin SysFS
+func PluginFS() SysFS {
+	return NewSysFS(filepath.Join(UserLibexecDir, "plugins"))
```
Comment (shizhMSFT) [inline dir/fs.go:49]: Let's make `plugins` a const.

File: dir/path.go [hunk 5]
```diff
@@ -1,88 +1,71 @@
 package dir
```
Comment (shizhMSFT) [inline dir/path.go:1]: It is better to document the package. like
```golang
// Package dir does bla bla bla...
package dir
```

File: dir/path_linux.go [hunk 6]
```diff
@@ -0,0 +1,15 @@
+//go:build linux
+// +build linux
```
Comment (shizhMSFT) [inline dir/path_linux.go:2]: We don't need to specify `linux`. This is considered as general implantation.

File: dir/path.go [hunk 7]
```diff
@@ -1,88 +1,71 @@
...
+func init() {
+	loadPath()
 }
 
-func checkError(err error) {
-	// if path does not exist, the path can be used to create file
-	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+// loadPath function defines the directory for
+// NotationLibexec, NotationConfig
+func loadPath() {
```
Comment (shizhMSFT) [inline dir/path.go:44]: How about renaming
```suggestion
func loadUserPaths() {
```

File: dir/path_darwin.go [hunk 8]
```diff
@@ -0,0 +1,15 @@
+//go:build darwin
+// +build darwin
```
Comment (shizhMSFT) [inline dir/path_darwin.go:2]: It is not necessary to specify go build flags. Files names with `_darwin.go` suffix are special ones similar to `_windows.go`, `_test.go` and etc.

File: dir/path_linux.go (hunk 6 above)
Comment (shizhMSFT) [reply dir/path_linux.go:2]: We probably need to rename the file name to `path_unix.go`.

Comment (shizhMSFT) [reply dir/path_linux.go:2]: Try reading this doc to learn more: https://www.digitalocean.com/community/tutorials/building-go-applications-for-different-operating-systems-and-architectures#using-goos-and-goarch-filename-suffixes

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: internal/mock/mock_fs/fs.go (hunk 1 above)
Comment (JeyJeyGao) [reply internal/mock/mock_fs/fs.go]: Renamed to mockfs.

File: dir/fs.go (hunk 2 above)
Comment (JeyJeyGao) [reply dir/fs.go]: Updated  `func NewSysFS(root string) SysFS`

File: dir/fs.go (hunk 4 above)
Comment (JeyJeyGao) [reply dir/fs.go]: Added.

File: dir/path.go (hunk 5 above)
Comment (JeyJeyGao) [reply dir/path.go:23]: Added doc.

File: dir/path_linux.go (hunk 6 above)
Comment (JeyJeyGao) [reply dir/path_linux.go]: Updated.

File: dir/path.go (hunk 7 above)
Comment (JeyJeyGao) [reply dir/path.go]: Updated.

File: dir/path_darwin.go [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+package dir
+
+func init() {
+	loadDarwinPath()
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply dir/path_darwin.go]: Updated.

File: dir/path_darwin.go (hunk 8 above)
Comment (JeyJeyGao) [reply dir/path_darwin.go]: Updated.  Learned a lot.

File: dir/path.go (hunk 3 above)
Comment (JeyJeyGao) [reply dir/path.go:40]: Updated.

File: dir/path.go [hunk 10]
```diff
@@ -1,88 +1,95 @@
...
 	TrustStoreDir = "truststore"
 )
 
-// PathManager contains the union directory file system and methods
-// to access paths of notation
-type PathManager struct {
-	ConfigFS     UnionDirFS
-	LibexecFS    UnionDirFS
-	UserConfigFS UnionDirFS
+var userConfigDir = os.UserCacheDir
```
Comment (priteshbandi) [inline dir/path.go:61]: Why are we defining userConfigDir variable?

**Review by priteshbandi — APPROVED** [review]
LGTM with one question

File: dir/path.go (hunk 10 above)
Comment (JeyJeyGao) [reply dir/path.go:61]: It's for unit test to replace the os.UserCacheDir() function with a mock function 😂

Comment (JeyJeyGao) [reply dir/path.go]: Added coments.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**